File: Lab 01/code/fib.py

#!/usr/bin/python
import timeit
import datetime as dt
import matplotlib.pyplot as plt
import numpy as np


def fib_r(x):
    if (x <= 2):
        return 1
    return fib_r(x-1) + fib_r(x-2)


def fib_i(x):
    a = 1
    b = 1
    fib = 1
    i = 2
    while i < x:
        fib = a + b
        a = b
        b = fib
        i += 1

    return fib


"""
# measure time for recursion
a = []
time = []
for x in range(1, 41):
    start_r = dt.datetime.now()
    print("Fib of " + str(x) + " = " + str(fib_r(x)))
    end_r = dt.datetime.now()
    elapsed_r = (end_r - start_r).microseconds
    elapsed_r = elapsed_r*1000
    time.append(elapsed_r)
    print("Python recursion, number =" + str(x) + " time = "+str(elapsed_r))



print(*time)
list1 = range(1, 41)
print(*list1)

plt.plot(list1, time)
plt.xlabel('number')
plt.ylabel('time taken (nanoseconds)')
plt.title('Fibbonacci Python implentation using recursion')
plt.grid(True)
plt.show()
"""
a = []
time = []
# measure time for iteration
for x in range(1, 41):
    start_i = dt.datetime.now()
    print("Fib of " + str(x) + " = " + str(fib_i(x)))
    end_i = dt.datetime.now()
    elapsed_i = (end_i - start_i).microseconds
    elapsed_i = elapsed_i*1000
    time.append(elapsed_i)
    print("Python iterations, number =" + str(x) + " time = "+str(elapsed_i))

print(*time)
list1 = range(1, 41)
print(*list1)

plt.plot(list1, time)
plt.xlabel('number')
plt.ylabel('time taken (nanoseconds)')
plt.title('Fibbonacci Python implentation using iteration')
plt.grid(True)
plt.show()

File: Lab 01/code/Fib.java
import java.util.Arrays;

class Fib {

    public static int fib_r(int x) {
	if(x <= 2) return 1;
	return fib_r(x-1) + fib_r(x-2);
    }

    public static int fib_i(int x) {
	int a=1, b=1, fib=1, i=2;
	while(i<x) {
	    fib = a + b;
	    a = b;
	    b = fib;
	    i+=1;
	}
	return fib;
    }


    public static void main(String [] args) {
	int x = 10;
	int a[]=new int[40];
	float time[]=new float[40];


	/*for(int i=1;i<41;i++) {
		//measure time for executing using recursion java
		long startTime_r = System.nanoTime();
		System.out.println("Fib of " + i + " = " + fib_r(i));
		long stopTime_r = System.nanoTime();
		long elapsedTime_r = stopTime_r - startTime_r;
		a[i-1]=i;
		time[i-1]=elapsedTime_r;
		System.out.println("Java recursion time, number = " + i + " time in (nano sec) = " + elapsedTime_r);

	}

		System.out.println(Arrays.toString(time));*/

		//measure time for executing using iteration java
		for(int j=1;j<41;j++) {
			long startTime_i = System.nanoTime();
			System.out.println("Fib of " + j + " = " + fib_i(j));
			long stopTime_i = System.nanoTime();
			long elapsedTime_i = stopTime_i - startTime_i;
			a[j-1]=j;
			time[j-1]=elapsedTime_i;
			System.out.println("Java iteration time, number = " + j + " time in (nano sec) = " + elapsedTime_i);
		}

		System.out.println(Arrays.toString(time));
    }
}

	
